addons/web/static/src/search/search_panel/search_view.scss

```diff
@@ -148,62 +144,49 @@
     }
 }
 
+.o_add_buttons {
+    padding: $dropdown-item-padding-y $dropdown-item-padding-x;
+}
+
 .o_web_client .o_mobile_search {
-    align-items: normal;
-    background-color: var(--mobileSearch-bg, #{$o-white});
-    border: none;
-    bottom: 0;
-    display: flex;
-    flex-direction: column;
-    left: 0;
-    overflow: auto;
-    padding: 0;
-    position: fixed;
-    top: 0;
-    width: 100%;
+    top: $o-navbar-height-xs;
     z-index: $zindex-modal;
 
-    .o_mobile_search_header {
-        background-color: var(--mobileSearch__header-bg, #{$o-brand-odoo});
-        display: flex;
-        min-height: $o-navbar-height;
-        justify-content: space-between;
-        width: 100%;
-
-        .o_mobile_search_button {
-            color: white;
-
-            &:active {
-                background-color: darken($o-brand-primary, 10%);
-            }
-        }
+    .o_searchview{
+        margin: 0 !important;
+        padding: 0 !important;
+        border: 0 !important;
     }
 
     .o_mobile_search_content {
         flex: auto;
+        width: 100%;
         height: auto;
         overflow-y: auto;
-        width: 100%;
 
         .o_searchview_input_container {
-            display: flex;
-            padding: 15px 20px 0 20px;
             position: relative;
+            display: flex;
+            flex-direction: column-reverse;
+            background-color: $o-gray-100;
     
             .o_searchview_input {
-                border-bottom: 1px solid $o-brand-secondary;
-                margin-bottom: 15px;
+                @extend .form-control-lg;
```

Comment:
No, never.
It looks handy but it will deeply impact compilation-time, especially since:

1. You're extending a default class. It's probably called in 100 other places, this will create  a very complex selector.
2. You're extending into a (very)nested selector,  `.o_web_client .o_mobile_search .o_mobile_search_content . o_searchview_input_container . o_searchview_input { }`. This will exponentially complicate the selector even more.  

We use @extend with pseudo-classes only and within the same context.
more https://sass-lang.com/documentation/at-rules/extend